vividus/src/main/java/org/vividus/runner/DeprecatedStepsReplacer.java

```diff
@@ -0,0 +1,231 @@
+
+                    if (!deprecatedStepFormatPattern.isEmpty())
+                    {
+                        try (Formatter formatter = new Formatter())
+                        {
+                            formatter.format(deprecatedStepFormatPattern, (Object[]) stepParameters);
+                            String escapedDeprecatedStep = Pattern.quote(step);
+                            String escapedActualStep = Matcher.quoteReplacement(formatter.toString());
+                            processedContent = nestingLevel == 0
+                                    ? processedContent.replaceFirst(escapedDeprecatedStep + END_OF_STEP_LOOKAHEAD,
+                                            escapedActualStep)
+                                    : replaceInTable(processedContent, step, escapedActualStep, nestingLevel);
+                        }
+                    }
+                    break;
+                }
+            }
+        }
+        return currentLevelNestedSteps.isEmpty()
+                ? processedContent : replaceSteps(processedContent, currentLevelNestedSteps, ++currentNestingLevel);
+    }
+
+    private String getActualStepFormatPattern(StepCandidate stepCandidate)
+    {
+        Optional<String> deprecationPatternInComposite = Stream.of(stepCandidate.composedSteps()).flatMap(
+                c -> DeprecatedCompositeStepsReporter.DEPRECATED_COMPOSITE_STEP_COMMENT_PATTERN.matcher(c).results())
+                .map(r -> r.group(2)).findFirst();
+        return deprecationPatternInComposite.orElseGet(
+                () -> Optional.ofNullable(stepCandidate.getMethod())
+                        .map(m -> stepCandidate.getMethod().getAnnotation(Replacement.class))
+                        .map(Replacement::replacementFormatPattern).orElse(StringUtils.EMPTY));
+    }
+
+    private String[] getParametersFromStep(Matcher matcher, List<String> nestedSteps)
+    {
+        matcher.matches();
+        String[] parameters = new String[matcher.groupCount()];
+        for (int i = 0; i < parameters.length; i++)
+        {
+            parameters[i] = matcher.group(i + 1);
+            fetchNestedStepsFromParameter(nestedSteps, parameters[i]);
+        }
+        return parameters;
+    }
+
+    private void fetchNestedStepsFromParameter(List<String> nestedSteps, String parameter)
+    {
+        Matcher matcher = NESTED_STEPS_TABLE_PATTERN.matcher(parameter);
+        if (matcher.find())
+        {
+            ExamplesTableFactory examplesTableFactory = configuration.examplesTableFactory();
+            ExamplesTable examplesTable = examplesTableFactory.createExamplesTable(parameter);
+
+            String tableContent = examplesTable.getRows().stream().map(p -> p.get(MANDATORY_STEP_COLUMN_NAME))
+                    .collect(Collectors.joining(System.lineSeparator()));
+            List<String> tableNestedSteps = configuration.storyParser().parseStory(tableContent).getScenarios().get(0)
```

Comment:
I see some similarities to `StepExamplesTableParser`

Reply:
> I see some similarities to `StepExamplesTableParser`

Yep, I took the logic from there partially
But method of the class return `Step` objects and we can't reuse it